src/shared/parameter_v2/config/examples/generated_c_test.cpp

```diff
@@ -0,0 +1,61 @@
+#include <gtest/gtest.h>
+
+#include "shared/parameter_v2/config/examples/example_config.h"
+#include "shared/parameter_v2/config/examples/parameter.h"
+
+TEST(BasicAccess, test_basic_parameter_access)
+{
+    const ThunderbotsConfig_t* tbots_config = app_dynamic_parameters_create();
+
+    ASSERT_TRUE(tbots_config->ExampleConfig->FooConfig->foo_bool.value == true);
+    ASSERT_TRUE(tbots_config->ExampleConfig->BarConfig->bar_bool.value == true);
+    ASSERT_TRUE(tbots_config->ExampleConfig->FooConfig->foo_int.value == 3);
+    ASSERT_TRUE(tbots_config->ExampleConfig->BarConfig->bar_int.value == 3);
+
+    // WONT COMPILE
+    // tbots_config->ExampleConfig->BarConfig->bar_bool.value = true;
+    // tbots_config->ExampleConfig->FooConfig->foo_int.value = 3;
+}
+
+TEST(BasicAccess, test_that_we_can_test_with_this_config)
+{
+    FooConfig_t foo_config = {
+        .foo_int  = {.value = 420},
+        .foo_bool = {.value = false},
+    };
+
+    // can pass foo_config into a test that needs it
+    ASSERT_TRUE(foo_config.foo_bool.value == false);
+    ASSERT_TRUE(foo_config.foo_int.value == 420);
+}
+
+TEST(BasicAccess, test_that_we_can_test_with_nested_configs)
```

Comment:
I'm a bit confused about the `test_that_we_can_test` prefix on these tests. Could you elaborate a bit?